pkg/apis/servicecatalog/validation/instance.go

```diff
@@ -308,22 +351,46 @@ func internalValidateServiceInstanceStatusUpdateAllowed(new *sc.ServiceInstance,
 }
 
 func internalValidateServiceInstanceReferencesUpdateAllowed(new *sc.ServiceInstance, old *sc.ServiceInstance) field.ErrorList {
+	var errMsg string
 	allErrs := field.ErrorList{}
+
 	if new.Status.CurrentOperation != "" {
 		allErrs = append(allErrs, field.Forbidden(field.NewPath("status").Child("currentOperation"), "cannot update references when currentOperation is present"))
 	}
-	if new.Spec.ClusterServiceClassRef == nil {
-		allErrs = append(allErrs, field.Required(field.NewPath("spec").Child("clusterServiceClassRef"), "clusterServiceClassRef is required when updating references"))
+
+	if new.Spec.ClusterServiceClassRef == nil && new.Spec.ServiceClassRef == nil {
+		errMsg = "clusterServiceClassRef or serviceClassRef is required when updating references"
+		allErrs = append(allErrs, field.Required(field.NewPath("spec").Child("clusterServiceClassRef"), errMsg))
+		allErrs = append(allErrs, field.Required(field.NewPath("spec").Child("serviceClassRef"), errMsg))
+	}
+	if new.Spec.ClusterServicePlanRef == nil && new.Spec.ServicePlanRef == nil {
+		errMsg = "clusterServicePlanRef or servicePlanRef is required when updating references"
+		allErrs = append(allErrs, field.Required(field.NewPath("spec").Child("clusterServicePlanRef"), errMsg))
+		allErrs = append(allErrs, field.Required(field.NewPath("spec").Child("servicePlanRef"), errMsg))
 	}
-	if new.Spec.ClusterServicePlanRef == nil {
-		allErrs = append(allErrs, field.Required(field.NewPath("spec").Child("clusterServicePlanRef"), "clusterServicePlanRef is required when updating references"))
+	if new.Spec.ClusterServiceClassRef != nil && new.Spec.ServiceClassRef != nil {
+		errMsg = "clusterServiceClassRef and serviceClassRef cannot both be set when updating references"
+		allErrs = append(allErrs, field.Required(field.NewPath("spec").Child("clusterServiceClassRef"), errMsg))
```

Comment:
should lines 373 and 374 be invalids, not required?

Reply:
yep! Don't remember why I chose Required here.